plugin/PlayAssetDelivery/src/test/java/com/google/play/core/godot/assetpacks/PlayAssetDeliveryTest.java

```diff
@@ -42,8 +43,28 @@
   @Mock Godot godotMock;
   @Mock AssetPackManager assetPackManagerMock;
 
+  private String receivedSignalName;
+  private Object[] receivedSignalObjects;
+
+  /**
+   * Creates a mock PlayAssetDelivery instance. Overrides the emitSignal method inherited from
+   * GodotPlugin. In this way whenever emitSignal() is called, we can assert the side-effects it
+   * resulted on receivedSignalName and receivedSignalObjects variables.
+   */
   private PlayAssetDelivery createPlayAssetDeliveryInstance() {
-    return new PlayAssetDelivery(godotMock, assetPackManagerMock);
+    return new PlayAssetDelivery(godotMock, assetPackManagerMock) {
```

Comment:
You can make PlayAssetDelivery into a [Spy](https://javadoc.io/static/org.mockito/mockito-core/3.3.3/org/mockito/Spy.html) and use [ArgumentCaptor](https://site.mockito.org/javadoc/current/org/mockito/ArgumentCaptor.html) to intercept the signalName and args.

Reply:
Used ArgumentCaptor approach, created a emitSignalWrapper() package private method in PlayAssetDelivery since emitSignal is protected.